src/business_logic/execution/execution_entry_point.rs

```diff
@@ -265,9 +264,9 @@ impl ExecutionEntryPoint {
         &self,
         state: &mut S,
     ) -> Result<[u8; 32], ExecutionError> {
-        if self.class_hash.is_some() {
+        if self.class_hash != *UNINITIALIZED_CLASS_HASH {
             match self.call_type {
-                CallType::Delegate => return Ok(self.class_hash.unwrap()),
+                CallType::Delegate => return Ok(self.class_hash),
                 _ => return Err(ExecutionError::CallTypeIsNotDelegate),
             }
```

Comment:
I dont understand in what case we would have `self.class_hash == UNINITIALIZED_CLASS_HASH`